arrow2_convert/src/deserialize.rs

```diff
@@ -71,11 +88,11 @@ macro_rules! impl_arrow_array {
         impl ArrowArray for $array {
             type BaseArrayType = Self;
 
-            fn iter_from_array_ref(b: &dyn Array) -> <&Self as IntoIterator>::IntoIter {
+            fn iter_from_array_ref(b: &dyn Array) -> <Self as IterRef>::Iter<'_> {
                 b.as_any()
                     .downcast_ref::<Self::BaseArrayType>()
                     .unwrap()
```

Comment:
@ncpenke I was wondering about this unwrap - can it panic?... This doesn't look too nice

Reply:
@aldanor In theory it can, though we do a top-level schema check. But if the underlying arrow structure is different than the schema it advertises, then we can panic. I'll fix it in my upcoming change. I'll add you as a reviewer so that you can include any other feedback.

Reply:
@aldanor Just a heads up another thing we need to do is introduce our own error type (https://github.com/DataEngineeringLabs/arrow2-convert/issues/54). We're currently piggybacking-off arrow2's type.

Reply:
Yea, I can take on that (the error type).

Reply:
Thanks! I'm almost done with my changes. I just changed the unwrap to an Option, and will leave the error type change to you.